trestle/core/markdown/control_markdown_node.py

```diff
@@ -0,0 +1,464 @@
+                            'However, none was found.'
+                        )
+                    content.objective_part = common.Part(
+                        name='objective', id=f'{tree_context.control_id}_obj', prose=''
+                    )
+
+                i = self._process_part_line(i, line, lines, content.objective_part)
+
+            elif is_control_guidance:
+                # Read control guidance to a part object
+                if not content.guidance_part:
+                    if not tree_context.control_id:
+                        raise TrestleError(
+                            'Unexpected error, control id, group and title should be before ## Control guidance.'
+                            'However, none was found.'
+                        )
+
+                    guidance_id = ControlInterface.strip_to_make_ncname(tree_context.control_id + '_gdn')
+                    content.guidance_part = common.Part(id=guidance_id, name=control_md_heading_label_ncname, prose='')
+
+                i = self._process_part_line(i, line, lines, content.guidance_part, read_parts=False)
+
+            elif is_generic_control_part:
+                # Read other control parts to a part objects
+                if not content.other_parts:
+                    if not tree_context.control_id:
+                        raise TrestleError(
+                            'Unexpected error, control id, group and title should be before ## Control parts.'
+                            'However, none was found.'
+                        )
+
+                    part_id = ControlInterface.strip_to_make_ncname(
+                        tree_context.control_id + '_' + control_md_heading_label
+                    )
+                    a_part = common.Part(id=part_id, name=control_md_heading_label_ncname, prose='')
+                    content.other_parts.append(a_part)
+
+                i = self._process_part_line(i, line, lines, content.other_parts[-1], read_parts=False)
+            else:
+                i += 1
+
+        first_line_to_grab = starting_line - 1 if starting_line else 0
+        content.raw_text = '\n'.join(lines[first_line_to_grab:i])
+
+        def _strip_prose_or_none(part: Optional[common.Part]):
+            if part and part.prose:
+                part.prose = part.prose.strip() if part.prose.strip() else None
+
+        _strip_prose_or_none(content.statement_part)
+        _strip_prose_or_none(content.objective_part)
+        _strip_prose_or_none(content.guidance_part)
+        for some_part in content.other_parts:
+            _strip_prose_or_none(some_part)
+
+        md_node = ControlMarkdownNode(key=root_key, content=content, starting_line=first_line_to_grab)
+        md_node.subnodes = node_children
+        return md_node, i
+
+    def get_control_statement(self) -> Optional[ControlMarkdownNode]:
+        """Get control statement node."""
+        return self.get_node_for_key('## Control Statement')
+
```

Comment:
Are we clear and consistent on capitalization of Statement and Guidance?

Reply:
I decided to make matching case insensitive so it does not matter which way it is written